pkg/kubelet/cm/dra/plugin/client.go

```diff
@@ -42,13 +52,58 @@ func NewDRAPluginClient(pluginName string) (drapb.NodeClient, error) {
 	return existingPlugin, nil
 }
 
-func (p *plugin) NodePrepareResources(
+type Plugin struct {
+	ctx    context.Context
```

Comment:
Why are we storing a `context.Context`? This is [generally discouraged](https://go.dev/blog/context-and-structs), as the context should be ephemeral state only used in a typically short-lived call chain.

Reply:
The plugin has a context that applies to all gRPC connections, independent from the specific gRPC call that causes those connections to be created. Therefore `getOrCreateGRPCConn` needs access to both that long-lived plugin context.

In general, context handling in gRPC is a mess :cry:

Reply:
This change is also part of https://github.com/kubernetes/kubernetes/pull/125163/files - that PR would be a better place to continue this discussion, if needed.